exchange/seat_non_bids.go

```diff
@@ -53,3 +58,55 @@ func (snb *nonBids) get() []openrtb_ext.SeatNonBid {
 	}
 	return seatNonBid
 }
+
+// newProxyNonBid creates a new proxy non-bid object with the given impression ID and non-bid reason.
+func newProxyNonBid(impId string, nonBidReason int) openrtb_ext.NonBid {
+	return openrtb_ext.NonBid{
+		ImpId:      impId,
+		StatusCode: nonBidReason,
+	}
+}
+
+// buildProxyNonBids creates a list of proxy non-bids with the given impression IDs and non-bid reason.
+// this method is not thread safe as we are initializing and writing to map
+func buildProxyNonBids(impIds []string, nonBidReason openrtb3.NoBidReason) []openrtb_ext.NonBid {
+	if len(impIds) == 0 {
+		return nil
+	}
+	proxyNonBids := []openrtb_ext.NonBid{}
+	for _, impId := range impIds {
+		nonBid := newProxyNonBid(impId, int(nonBidReason))
+		proxyNonBids = append(proxyNonBids, nonBid)
+	}
+	return proxyNonBids
+}
+
+// addProxyNonBids adds the proxy non-bids to the seatNonBidsMap.
+// It takes a list of impression IDs, a non-bid reason, and a seat as input parameters.
+// It builds the proxy non-bids using the buildProxyNonBids function and appends them to the seatNonBidsMap.
+// If the seatNonBidsMap is nil, it initializes it with an empty map.
+// The proxy non-bids are added to the seatNonBidsMap under the specified seat.
+// This method is not thread safe as we are initializing and writing to map
+func (snb *nonBids) addProxyNonBids(impIds []string, nonBidReason openrtb3.NoBidReason, seat string) {
+	if len(impIds) == 0 {
+		return
+	}
+	proxyNonBids := buildProxyNonBids(impIds, nonBidReason)
+	if snb.seatNonBidsMap == nil {
+		snb.seatNonBidsMap = make(map[string][]openrtb_ext.NonBid)
+	}
+	snb.seatNonBidsMap[seat] = append(snb.seatNonBidsMap[seat], proxyNonBids...)
+}
+
+// append adds the nonBids from the input nonBids to the current nonBids.
+// This method is not thread safe as we are initializing and writing to map
+func (snb *nonBids) append(nonBids ...nonBids) {
```

Comment:
The similarity to the built in `append` function is confusing. Why do we need to append instead of adding to the builder directly?